source/common/network/address_impl.cc

```diff
@@ -130,7 +139,15 @@ int Ipv4Instance::connect(int fd) const {
 }
 
 int Ipv4Instance::socket(SocketType type) const {
-  return ::socket(AF_INET, flagsFromSocketType(type), 0);
+  int fd = ::socket(AF_INET, flagsFromSocketType(type), 0);
+  RELEASE_ASSERT(fd != -1);
+
+#ifdef __APPLE__
+  // Cannot set SOCK_NONBLOCK as a ::socket flag
```

Comment:
I switched flagsFromSocketType to be socketFromSocketType and do all the flag setting, fcntl, and socket creation there. I think this is better, since it's all contained in one block.